js/src/block-editor/components/edit.js

```diff
@@ -26,16 +28,42 @@ import { EDITOR_LOCATION } from '../../common/constants';
  * @return {React.ReactElement} The editor display.
  */
 const Edit = ( { blockProps, block } ) => {
-	const { attributes, className, isSelected } = blockProps;
+	const { attributes, className, clientId, isSelected } = blockProps;
 	const hasEditorField = getFieldsAsArray( block.fields ).some( ( field ) => {
 		return ! field.location || EDITOR_LOCATION === field.location;
 	} );
 
+	/**
+	 * Gets whether the passed block has a selected InnerBlock.
+	 *
+	 * @param {Object} blockCandidate The block to examine for InnerBlocks.
+	 * @param {Object} selectedBlock The block that's selected in the editor.
+	 * @return {boolean} Whether the passed block has a selected InnerBlock.
+	 */
+	const hasSelectedInnerBlock = ( blockCandidate, selectedBlock ) => {
+		return blockCandidate?.innerBlocks?.length &&
+			blockCandidate.innerBlocks.some( ( innerBlock ) =>
+				( Boolean( selectedBlock?.clientId ) && innerBlock?.clientId === selectedBlock?.clientId ) ||
+				hasSelectedInnerBlock( innerBlock, selectedBlock )
+			);
+	};
+
+	/** @type {boolean} Whether this block has an inner block that's selected. */
+	const isInnerBlockSelected = useSelect(
+		( select ) => {
+			const store = select( blockEditorStore );
+
+			// @ts-ignore Type definition is outdated.
+			return hasSelectedInnerBlock( store.getBlock( clientId ), store.getSelectedBlock() );
+		},
+		[ clientId, isSelected ]
+	);
+
 	return (
 		<>
 			<GcbInspector blockProps={ blockProps } block={ block } />
 			<div className={ className } key={ `form-controls-${ block.name }` }>
-				{ isSelected && hasEditorField ? (
+				{ ( isSelected || isInnerBlockSelected ) && hasEditorField ? (
```

Comment:
This is needed because the block's prop `isSelected` is false if one if its InnerBlocks is selected.

So on clicking an InnerBlock, it changed the focus to the `<ServerSideRender>`, and you couldn't edit it:

![on-changing-](https://user-images.githubusercontent.com/4063887/123906146-417ce700-d939-11eb-87d8-42d4946a3052.gif)